### PR TITLE
chore: Update GitHub actions to use Node 16 and Ubuntu Latest

### PR DESCRIPTION
**Pull request by SDawley**
Signed-off-by: sdawley <sdawley@redhat.com>

### What does this PR do?
Update GitHub actions to use Node 16 and Ubuntu Latest

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/21763

### How to test this PR?

Run GH Actions to make sure they still work properly.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [ ] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: .github/workflows/publish-pr-check-content.yml [hunk 1]
```diff
@@ -20,16 +20,16 @@ jobs:
     # only if it's a pull request
     if: github.event.workflow_run.event == 'pull_request'
     name: publish
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-latest
     steps:
       - name: download dist artifact
-        uses: dawidd6/action-download-artifact@v2
+        uses: dawidd6/action-download-artifact@v2.18.0
```
Comment (nickboldt) [inline .github/workflows/publish-pr-check-content.yml:26]: Why not use https://github.com/dawidd6/action-download-artifact/releases/tag/v2.24.2 if you're going to hardcode a specific v2 version?

Also I don't know the answer to this, but will GH pull the latest 2.y.z if you set @v2, or do you have to set the specific version of 2.y.z to get that?

**Comment by nickboldt** [comment]
> @SDawley please fix shellcheck GH action's failures https://github.com/eclipse-che/che-plugin-registry/actions/runs/3652981131/jobs/6179520892

To be fair, this PR is for yaml files, not bash files... do we really need a passing shellcheck for yaml? :D Maybe a second PR would be better for that...

File: .github/workflows/publish-pr-check-content.yml (hunk 1 above)
Comment (SDawley) [reply .github/workflows/publish-pr-check-content.yml]: Looking at https://github.com/actions/upload-artifact as an example it looks like
https://github.com/actions/upload-artifact/tree/v3.1.1 matches https://github.com/actions/upload-artifact/tree/v3

so yes v3  gets the latest v3.x.y. Should I hardcode it to the latest 3 digit version for traceability purposes?

Also re: why I used 2.18.0 I think my original intent was to use the first version that switched to node 16 so there would be the smallest amount of change/impact/potential breakage.

Comment (nickboldt) [reply .github/workflows/publish-pr-check-content.yml]: I'm torn between the traceability of `vx.y.z` versions and the automatic updating of `vx`. 

Given we use `vx` versions today and have rarely had unexpected build failures... I think it's safe to keep that up. But if some day we decide to build a tool to fetch latest versions and update GH actions w/ `vx.y.z` and keep it current, we could switch then to the more declarative system.

**Comment by svor** [comment]
> To be fair, this PR is for yaml files, not bash files... do we really need a passing shellcheck for yaml? :D Maybe a second PR would be better for that...

yes, but before these changes shellcheck PR check was passed without errors (see previous PRs), probably because of https://github.com/eclipse-che/che-plugin-registry/pull/1506/files#diff-3c73f64339a28ea5cd74f18201c12edf164e925b8816dbacd64c0116c6671399R16

I'm OK to have the fix for ShellCheck in another PR, but let's revert changes in [shellcheck-pr-check.yml](https://github.com/eclipse-che/che-plugin-registry/pull/1506/files#diff-3c73f64339a28ea5cd74f18201c12edf164e925b8816dbacd64c0116c6671399) in this PR

**Comment by SDawley** [comment]
I reverted shellcheck to Ubuntu 20.04 and it works now, though we should figure out why it breaks on 22.04.

**Review by svor — APPROVED** [review]

File: .github/workflows/publish-plugin-report.yaml [hunk 2]
```diff
@@ -29,7 +29,7 @@ jobs:
       with:
         node-version: '16'
     - name: Get current time
-      uses: 1466587594/get-current-time@v2
+      uses: 1466587594/get-current-time@v2.0.2
```
Comment (nickboldt) [inline .github/workflows/publish-plugin-report.yaml:32]:
```suggestion
      uses: 1466587594/get-current-time@v2
```

**Review by svor — APPROVED** [review]

**Pull request merged**